README.md

```diff
@@ -42,49 +42,53 @@ in sync with the written specification on conventionalcommits.org.
-                 |  <summary>, <newline>*, <footer>+
-                 |  <summary>, <newline>*
-
-<summary>       ::= <type>, "(", <scope>, ")", ["!"], ":", <whitespace>*, <text>
-                 |  <type>, ["!"], ":", <whitespace>*, <text>
-<type>          ::= <any UTF8-octets except newline or parens or ":" or "!:" or whitespace>+
-<scope>         ::= <any UTF8-octets except newline or parens>+
-<text>          ::= <any UTF8-octets except newline>*
-
-/*
- * Note: if the first <body> node starts with "BREAKING CHANGE:" this should
- * be treated by parsers as a breaking change marker upstream:
- */
-<body>          ::= [<any text except pre-footer>], <newline>, <body>*
-                 | [<any text except pre-footer>]
-/* Note: <pre-footer> is used during parsing, but never returned in the AST. */
-<pre-footer>    ::= <newline>*, <footer>+
-
-<footer>        ::= <token>, <separator>, <whitespace>*, <value>, [<newline>]
-<token>         ::= "BREAKING CHANGE"
-                 |  <type>, "(" <scope> ")", ["!"]
-                 |  <type>, ["!"]
-<separator>     ::= ":" | " #"
-<value>         ::= <text>, <continuation>+
-                 | <text>
-<continuation> ::= <newline>, <whitespace>+, <text>
+<whitespace>      ::= <ZWNBSP> | <TAB> | <VT> | <FF> | <SP> | <NBSP> | <USP>
+
+<message>         ::= <summary>, <newline>+, <body>, <newline>*, <footer>+
+                   |  <summary>, <newline>*, <footer>+
+                   |  <summary>, <newline>*
+
+/* "!" should be added to the AST as a <breaking-change> node with the value "!" */
+<summary>         ::= <type>, "(", <scope>, ")", ["!"], ":", <whitespace>*, <text>
+                   |  <type>, ["!"], ":", <whitespace>*, <text>
+<type>            ::= <any UTF8-octets except newline or parens or ":" or "!:" or whitespace>+
+<scope>           ::= <any UTF8-octets except newline or parens>+
+<text>            ::= <any UTF8-octets except newline>*
+
+
+<body>            ::= [<any body-text except pre-footer>], <newline>, <body>*
+                   |  [<any body-text except pre-footer>]
+/* For convenience the <breaking-change>, <separator>, <whitespace>, and
+ * <text> tokens of <body-text> should be appended as children to <body> */
+<body-text>       ::= [<breaking-change>, ":", <whitespace>*], text
+/* Note: <pre-footer> is used during parsing, but not returned in the AST. */
+<pre-footer>      ::= <newline>*, <footer>+
+
+<footer>          ::= <token>, <separator>, <whitespace>*, <value>, [<newline>]
+/* "!" should be added to the AST as a <breaking-change> node with the value "!" */
+<token>           ::= <breaking-change>
+                   |  <type>, "(" <scope> ")", ["!"]
+                   |  <type>, ["!"]
+<separator>       ::= ":" | " #"
+<value>           ::= <text>, <continuation>+
+                   |  <text>
+<continuation>    ::= <newline>, <whitespace>+, <text>
+
+<breaking-change> ::= "BREAKING CHANGE" | "BREAKING-CHANGE"
```

Comment:
The text on conventionalcommits.org indicates that we should support `BREAKING-CHANGE`.